### DISCUSSION TITLE
possible bug in ingest v0.2.13 ?

### DISCUSSION THREAD

**Discussion by blankdots** [neicnordic/sensitive-data-archive#734] [General]
in version v0.2.13 there seems to be a bug with
```
time="2024-03-08T06:54:17Z" level=error msg="could not find matching public key header, decryption failed"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x9e46e6]

goroutine 50 [running]:
main.main.func4()
        /go/cmd/ingest/ingest.go:366 +0x2946
created by main.main
        /go/cmd/ingest/ingest.go:99 +0x6db
```
was this fixed in later versions ?

**Comment by pontus** [comment]
The problem here is likely ingestion of files encrypted with an incorrect key (so that files are not decryptable with the keys available). 

The bug causing a segmentation fault seems to be at
https://github.com/neicnordic/sensitive-data-archive/blob/v0.2.13/sda/cmd/ingest/ingest.go#L359 (`err` is not scoped to the `if` so later when used, `err` is `nil` if there wasn't an MQ error).

It looks like these are fixed nowadays, https://github.com/neicnordic/sensitive-data-archive/blob/main/sda/cmd/ingest/ingest.go#L366.

**Reply by blankdots** [reply]
super thank you!
